includes/admin/extensions/class-extensions-api.php

```diff
@@ -0,0 +1,201 @@
+	 *
+	 * @since 2.11.x
+	 * @param array $body    The body for the API request.
+	 * @param int   $item_id The product ID, if querying a single product.
+	 * @return false|array|ProductData
+	 */
+	public function get_product_data( $body = array(), $item_id = false ) {
+		if ( empty( $body ) ) {
+			if ( empty( $item_id ) ) {
+				return false;
+			}
+			$body = $this->get_api_body( $item_id );
+		}
+		$key = $this->array_key_first( $body );
+		// The option name is created from the first key/value pair of the API "body".
+		$option_name = sanitize_key( "edd_extension_{$key}_{$body[ $key ]}_data" );
+		$option      = get_option( $option_name );
+		$is_stale    = ! empty( $option['timeout'] ) && time() > $option['timeout'];
+
+		// The ProductData class.
+		$product_data = new ProductData();
+
+		// If the data is "fresh" and what we want exists, return it.
+		if ( ! $is_stale ) {
+			if ( $item_id && ! empty( $option[ $item_id ] ) ) {
+				return $product_data->fromArray( $option[ $item_id ] );
+			} elseif ( ! empty( $option['timeout'] ) ) {
+				unset( $option['timeout'] );
+
+				return $option;
+			}
+		}
+
+		// Get all of the product data.
+		$all_product_data = $this->get_all_product_data();
+
+		// If no product data was retrieved, let the option sit for an hour.
+		if ( ! $all_product_data ) {
+			$data = array(
+				'timeout' => strtotime( '+1 hour', time() ),
+			);
+			if ( $option && $is_stale ) {
+				$data = array_merge( $option, $data );
+			}
+			update_option(
+				$option_name,
+				$data,
+				false
+			);
+
+			if ( $item_id && ! empty( $option[ $item_id ] ) ) {
+				return $product_data->fromArray( $option[ $item_id ] );
+			}
+			unset( $option['timeout'] );
+
+			return $option;
+		}
+
+		$value = array(
+			'timeout' => strtotime( '+1 week', time() ),
+		);
+		if ( $item_id && ! empty( $all_product_data->$item_id ) ) {
+			$item              = $all_product_data->$item_id;
+			$value[ $item_id ] = $this->get_item_data( $item );
+		} elseif ( in_array( $key, array( 'category', 'tag' ), true ) ) {
+			$term_id = $body[ $key ];
+			foreach ( $all_product_data as $item_id => $item ) {
+				if ( 'category' === $key && ( empty( $item->categories ) || ! in_array( $term_id, $item->categories, true ) ) ) {
+					continue;
+				} elseif ( 'tag' === $key && ( empty( $item->tags ) || ! in_array( $term_id, $item->tags, true ) ) ) {
+					continue;
+				}
+				$value[ $item_id ] = $this->get_item_data( $item );
+			}
+		}
+
+		update_option( $option_name, $value, false );
+		unset( $value['timeout'] );
+
+		return $item_id && ! empty( $value[ $item_id ] ) ? $product_data->fromArray( $value[ $item_id ] ) : $value;
+	}
+
+	/**
+	 * Gets all of the product data, either from a tranient or an API request.
+	 *
+	 * @since 2.11.x
+	 * @return object|false
+	 */
+	private function get_all_product_data() {
+		// Possibly all product data is in a transient. If it is, return it.
+		$all_product_data = get_transient( 'edd_all_extension_data' );
```

Comment:
Just curious if there was a reason for this one to use a transient when the other uses options. I think we tend to prefer options at least when making API requests to EDD's site because of annoyances we've had with transients in the past (Software Licensing).

Reply:
I like transients because I don't have to manage the expiration and I'm a little bit lazy? I can switch if you want, but I like transients, I confess.